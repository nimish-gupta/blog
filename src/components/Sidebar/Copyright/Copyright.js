import React from 'react';
import styled from 'styled-components';
import { lighten } from 'polished';

type Props = {
	copyright: string,
};

const StyledDiv = styled.div`
	color: ${(props) => lighten(0.18, props.theme.color.gray)};
	font-size: ${(props) => props.theme.typographic.smallFontSize};
`;

const Copyright = ({ copyright }: Props) => <StyledDiv>{copyright}</StyledDiv>;

export default Copyright;
