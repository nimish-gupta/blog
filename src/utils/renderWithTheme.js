import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={theme}>{component}</ThemeProvider>
	);
}
