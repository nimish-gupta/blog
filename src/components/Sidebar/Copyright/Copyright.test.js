// @flow strict
import React from 'react';
import renderWithTheme from '../../../utils/renderWithTheme';
import Copyright from './Copyright';

describe('Copyright', () => {
	it('renders correctly', () => {
		const props = {
			copyright: 'copyright',
		};

		const tree = renderWithTheme(<Copyright {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
