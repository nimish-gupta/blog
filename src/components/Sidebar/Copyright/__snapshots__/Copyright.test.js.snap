// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright renders correctly 1`] = `
<div
  className="sc-AxjAm fzHIaf"
>
  copyright
</div>
`;
